elasticsearch/templates/service.yaml

```diff
@@ -26,11 +34,10 @@ metadata:
     release: {{ .Release.Name | quote }}
     chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
     app: "{{ template "uname" . }}"
-  annotations:
-    # Create endpoints also if the related pod isn't ready
-    service.alpha.kubernetes.io/tolerate-unready-endpoints: "true"
 spec:
   clusterIP: None # This is needed for statefulset hostnames like elasticsearch-0 to resolve
+  # Create endpoints also if the related pod isn't ready
+  publishNotReadyAddresses: true
```

Comment:
This was added in [1.8](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG-1.8.md) so looks like this is the right time to switch 👍